### DISCUSSION TITLE
Setting Default Parameters for `chezmoiexternal`

### DISCUSSION THREAD

**Discussion by ProfessorManhattan** [twpayne/chezmoi#3376] [Q&A]
Hey, in my `chezmoiexternal` file, I have a lot of `git-repo` entries that have:

```
    clone.args = ["--depth", "1"]
    pull.args = ["--ff-only"]
```

Is there a way I can make it so that these are the default options so I can save a few hundred lines?

**Comment by twpayne** [comment]
`.chezmoiexternal` files are templates so you can use a template variable:

```
{{ $gitArgs := "clone.args = [\"--depth\", \"1\"]\npull.args = [\"--ff-only\"]" }}

[path/to/external]
    type = "git-repo"
    {{ $gitArgs | indent 4 }}
```

It's not particularly pretty, but it works.

**Comment by ameliaclark11** [comment]
default_git_args = ["--depth", "1"]

repo "example" {
    source = "https://github.com/example/repo.git"
    clone.args = default_git_args
    pull.args = ["--ff-only"]
}

# Other git-repo entries follow the same pattern with default_git_args
By using a variable like default_git_args, you can easily manage and update the default options for all your git-repo entries, reducing redundancy and saving lines. Now you can focus on more [activities in Lisbon](https://activitiesinportugal.com/activities-in-lisbon/) while enjoying a cleaner and more manageable chezmoiexternal file!
